core/src/main/scala/org/broadinstitute/dsde/rawls/metrics/RawlsInstrumented.scala

```diff
@@ -37,26 +43,48 @@ trait RawlsInstrumented extends DefaultInstrumented {
     */
   @implicitNotFound(msg = "Cannot expand instances of type ${A}")
   protected sealed trait Expansion[A] {
-    def makeName(key: String, a: A) = s"$key.${a.toString}"
+    def makeName(key: Option[String], a: A) = s"${getKey(key)}${a.toString}"
```

Comment:
So when we create stackdriver labels we need to give specify a `"label:key" "value"`. This gets translated as a little drop-down in the SD UI (here `instance_id` is the key):

![image](https://user-images.githubusercontent.com/5368863/28284565-6be1ff1e-6aff-11e7-9291-4b4c847f5f0c.png)

What I'm doing in my script is take a list of "known" keys (specified in a config in firecloud-develop) and parse out the values following those keys. So `workspace.<ws-id>.submission.<sub-id>` would be translated as 
```
label:workspace ws-id
label:submission sub-id
```
and we get the nice drop-down menus in SD.

I never said it was the most elegant thing in the world but that's how it's currently working...